File: plus-one/plus-one.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    let carry = 1;
    for (let i=digits.length-1;i > -1;i--) {
        let cur = digits[i];
        let calc = cur + carry;
        carry = 0;
        if (calc > 9) {
            calc = 0;
            carry = 1;
        }
        if (calc == digits[i]) {
            break;
        }
        digits[i] = calc;
    }
    if (carry == 1) {
        digits.unshift(1);
    }
    return digits;
};

File: plus-one/plus-one.java
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1;
        for (int i=digits.length-1; i>=0; i--) {
            if (digits[i]+carry >= 10) {
                digits[i] = digits[i]+carry-10;
                carry = 1;
            }
            else {
                digits[i] = digits[i]+carry;
                carry = 0;
            }
        }
        if (carry == 1) {
            int[] ans = new int[digits.length+1];
            ans[0] = 1;
            for (int i=0; i<digits.length; i++) {
                ans[i+1] = digits[i];
            }
            return ans;
        }
        else {
            return digits;
        }
    }
}